### ISSUE TITLE
MULTISET/JSON emulation can't read TIME type in Db2

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12158]
In Db2, the following query:

```sql
select cast(time '10:15:00' as varchar(10))
from sysibm.dual;
```

Produces

```
10.15.00
```

With `.` separators instead of `:`. We currently can't parse this format, and as such, can't deserialise `TIME` types contained in `MULTISET` operators, as they are serialised in JSON.

**Comment by lukaseder** [comment]
Given that Db2 has strong opinions here (see https://www.ibm.com/docs/en/db2-for-zos/11?topic=columns-date-time-timestamp-data-types), I guess we have to fix this in the `DefaultConverterProvider`, not in the JSON serialisation...

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.1 (https://github.com/jOOQ/jOOQ/issues/12159)
